src/sql-parser/src/parser.rs

```diff
@@ -1463,13 +1463,17 @@ impl<'a> Parser<'a> {
             Format::Regex(regex)
         } else if self.parse_keyword(CSV) {
             self.expect_keyword(WITH)?;
-            let (header_row, n_cols) = if self.parse_keyword(HEADER) || self.parse_keyword(HEADERS)
-            {
-                (true, None)
+            let columns = if self.parse_keyword(HEADER) || self.parse_keyword(HEADERS) {
+                if self.parse_keyword(COLUMNS) {
+                    let columns = self.parse_parenthesized_column_list(Mandatory)?;
+                    CsvColumns::HeaderNamed(columns)
+                } else {
+                    CsvColumns::Header
+                }
```

Comment:
I think this would be more in line with other similar SQL constructs if we dropped the `COLUMNS` word. So `CSV ... WITH HEADER` or `CSV ... WITH HEADER (a, b, c)`. And then this simplifies to
```
CsvColumns::Header {
    names: self.parse_parenthesized_column_list(Optional)?,
}
```

Reply:
done.